core-bundle/src/Resources/contao/widgets/RootPageDependentSelect.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao;
+
+class RootPageDependentSelect extends SelectMenu
+{
+	public function generate(): string
+	{
+		$fields = array();
+		$originalLabel = $this->strLabel;
+		$cssClasses = 'tl_select tl_chosen';
+		$rootPages = PageModel::findByPid(0, array('order' => 'sorting'));
+		$wizard = StringUtil::deserialize($this->wizard);
+
+		$this->blankOptionLabel = $GLOBALS['TL_LANG']['tl_module'][sprintf('%sBlankOptionLabel', $this->name)];
```

Comment:
What's this? Why not the regular one?

Reply:
The option to set a custom blank label instead of using the core default one (`-`):

https://github.com/contao/contao/blob/cf2ca71c614232865301443b71c66b6a31152a38/core-bundle/src/Resources/contao/library/Contao/Widget.php#L1360-L1364

It gives you an insight, what option you should select for: 

<img width="387" alt="Bildschirmfoto 2022-01-16 um 12 05 19" src="https://user-images.githubusercontent.com/754921/149657344-c96c6ef3-7880-481e-8e8a-007f23e2ddb0.png">

And think about having multiple `rootPageDependentSelect` widgets - you probably don't have the same blank label depending on your use case. This way you can fully customize it. Or is there another, more prefered way to achieve this? :-)

Reply:
as @Toflar is in holidays - WDYT @contao/developers?